smartsim/_core/_cli/build.py

```diff
@@ -248,14 +244,14 @@ def build_redis_ai(
 
         self.check_backends_install()
 
-        # Check for onnx and tf in user python environemnt and prompt user
-        # to download them if they are not installed. this should not break
-        # the build however, as we use onnx and tf directly from RAI instead
-        # of pip like we do PyTorch.
+        # Check for onnx and tf in user python environemnt or download
+        # them if requested. If they are not present and the user did not request
+        # they be downloaded, we still should not break the build, as we use
+        # onnx and tf directly from RAI instead of pip like we do PyTorch.
         if onnx:
-            self.check_onnx_install()
+            self.check_py_onnx_version(handle_conflict=self.modify_py_env)
         if tf:
-            self.check_tf_install()
+            self.check_py_tf_version(handle_conflict=self.modify_py_env)
```

Comment:
It looks like we might be repeating these `self.check_py_XXX_version(...)` calls in `Build.__init__` and `Build.install_redis_ai`

Can we do this once?

- [install_redis_ai](https://github.com/CrayLabs/SmartSim/blob/f812630adfa41a6caffe91615d35cc12e66c0023/smartsim/_core/_cli/build.py#L251-L271)

Reply:
Potentially? I am tempted to say yes, but it world required changing the order in which things are built (e.g. instead of building `redis -> torch -> rai` it would need to be `torch -> redis -> rai`).

Given the time to code freeze, and the fact that some others are currently working on this section of the code, I'm a little hesitant to squeze that change into this PR. Would you be okay with a follow on ticket?

Reply:
100% with you on either skipping or doing follow up. I didn't realize the order change would have to occur. Up to your discretion